### PR TITLE
BUG Fix: Allow using datetime64[ns, UTC] in IntervalDtype

### LINKED ISSUE #46999: BUG: Allow using datetime64[ns, UTC] in IntervalDtype
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
dates = pd.date_range("2022", periods=3, tz="UTC")
print(pd.IntervalIndex.from_arrays(dates[:-1], dates[1:]))
print(
    pd.IntervalIndex.from_arrays(
        ["2022-01-01", "2022-01-02"],
        ["2022-01-02", "2022-01-03"],
        dtype="interval[datetime64[ns, UTC], right]",
    )
)
```

### Issue Description

raises `TypeError: data type 'interval[datetime64[ns, UTC], right]' not understood`

xref #46666

works for Interval of naive timeseries...

```
dates = pd.date_range("2022", periods=3)
print(pd.IntervalIndex.from_arrays(dates[:-1], dates[1:]))
print(
    pd.IntervalIndex.from_arrays(
        ["2022-01-01", "2022-01-02"],
        ["2022-01-02", "2022-01-03"],
        dtype="interval[datetime64[ns], right]",
    )
)
```

### Expected Behavior

```
IntervalIndex([(2022-01-01, 2022-01-02], (2022-01-02, 2022-01-03]], dtype='interval[datetime64[ns, UTC], right]')
```

### Installed Versions

.

### PR DESCRIPTION
**Pull request by weikhor**
- [ ] closes #46999
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/arrays/interval/test_interval.py [hunk 1]
```diff
@@ -443,3 +443,15 @@ def test_arrow_interval_type_error_and_warning():
...
         ArrowIntervalType(pa.int64(), closed="both")
+
+
+def test_interval_index_subtype():
+    # GH 46999
+    dates = date_range("2022", periods=3, tz="UTC")
+    result = IntervalIndex.from_arrays(
+        ["2022-01-01", "2022-01-02"],
+        ["2022-01-02", "2022-01-03"],
+        dtype="interval[datetime64[ns, UTC], both]",
```
Comment (mroeschke) [inline pandas/tests/arrays/interval/test_interval.py:454]: 1. Could you parameterize over the `closed` options (I think there should be a fixture)
2. Could you also parameterize another timezone with UTC, e.g. `US/Pacific`
3. Will need a whatsnew note in v1.5

Comment (weikhor) [reply pandas/tests/arrays/interval/test_interval.py]: Have added parameterize over the closed options, parameterize another timezone and whatsnew note in v1.5

File: pandas/tests/arrays/interval/test_interval.py [hunk 2]
```diff
@@ -443,3 +443,19 @@ def test_arrow_interval_type_error_and_warning():
     msg = "Argument `closed` is deprecated in favor of `inclusive`"
     with tm.assert_produces_warning(FutureWarning, match=msg, check_stacklevel=False):
         ArrowIntervalType(pa.int64(), closed="both")
+
+
+@pytest.mark.parametrize("timezone", ["UTC", "US/Pacific", "GMT"])
+@pytest.mark.parametrize("inclusive", ["both", "neither", "left", "right"])
```
Comment (mroeschke) [inline pandas/tests/arrays/interval/test_interval.py:449]: Can you use the `closed` fixture instead?

File: doc/source/whatsnew/v1.5.0.rst [hunk 3]
```diff
@@ -756,7 +756,7 @@ Strings
 Interval
 ^^^^^^^^
 - Bug in :meth:`IntervalArray.__setitem__` when setting ``np.nan`` into an integer-backed array raising ``ValueError`` instead of ``TypeError`` (:issue:`45484`)
--
+- Bug in :class:`IntervalDtype` Allow using datetime64[ns, UTC] in IntervalDtype (:issue:`46999`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v1.5.0.rst:759]:
```suggestion
- Bug in :class:`IntervalDtype` when using datetime64[ns, tz] as a dtype string (:issue:`46999`)
```

File: pandas/tests/arrays/interval/test_interval.py (hunk 2 above)
Comment (jreback) [reply pandas/tests/arrays/interval/test_interval.py:449]: i think we just renamed that one right? (to inclusive) agreed should use the fixture

Comment (mroeschke) [reply pandas/tests/arrays/interval/test_interval.py:449]: Yeah or the `inclusive_endpoints_fixture`

Comment (weikhor) [reply pandas/tests/arrays/interval/test_interval.py:449]: Ok Have updated

**Review by mroeschke — APPROVED** [review]
LGTM (failures unrelated)

**Comment by mroeschke** [comment]
Thanks @weikhor!

**Pull request merged**